### PR TITLE
autodetect.go: combine two switch cases

### PR DESCRIPTION
**Pull request by TrumanLing**
Signed-off-by: Ling FaKe lingfake@huawei.com

### CODE REVIEW

File: image/autodetect.go [hunk 1]
```diff
@@ -61,8 +61,6 @@ func Autodetect(path string) (string, error) {
 
 	switch mimeType {
 	case "application/x-gzip":
-		return TypeImage, nil
-
 	case "application/octet-stream":
```
Comment (wking) [inline image/autodetect.go]: I don't think Go's `switch` supports fall-through without an [explict `fallthrough`][1].

[1]: https://golang.org/ref/spec#Fallthrough_statements

Comment (TrumanLing) [inline image/autodetect.go]: @wking It is no problem, I have verified it.

In generally, Go's `switch` supports style `case <con1>,<con2>:` and that I have submitted.

Comment (wking) [inline image/autodetect.go]: `case "application/x-gzip", "application/octet-stream":` would be fine, but what you have here is:

```go
case "application/x-gzip":
case "application/octet-stream":
```

And testing a similar layout locally:

```
$ cat x.go 
package main
import ("fmt")
func main() {
        var a = 0
        switch a {
        case 0:
        case 1:
                fmt.Println("hi")
        }
}
$ go run x.go 
```

I don't see `hi` in my stdout.

**Comment by TrumanLing** [comment]
@wking updated it, PTAL.

**Comment by wking** [comment]
8cc35783d looks good to me.

**Comment by coolljt0725** [comment]
LGTM

**Comment by vbatts** [comment]
LGTM

**Pull request merged**